model/DataObject.php

```diff
@@ -3403,6 +3404,29 @@ public function databaseIndexes() {
 			}
 		}
 
+		if ($sort && is_string($sort)) {
+		    $sort = preg_split('/,(?![^()]*+\\))/', $sort);
+
+			foreach ($sort as $value) {
```

Comment:
Please check out ArrayList::parseSortColumn(). This has pre-existing tests, and covers the range of various sort syntaxes a bit more robustly. For instance `$default_sort = '"SiteTree"."Order" DESC';` doesn't seem covered by this code.

Maybe we could extract that helper somewhere that index generation could access it?

Reply:
It looks like that method could also be extended to support multi-columns specifications; There's a chance to improve both ArrayList::sort() as well as index generation. :)

Reply:
I copied code from SQLSelect, happy to copy from somewhere else if it's more robust :)